lib/user.rb

```diff
@@ -0,0 +1,20 @@
+class User
+  attr_reader :name, :jokes
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << joke
+  end
+
+  def tell(person, joke)
+    person.jokes << joke
+  end
+
+  def joke_by_id(id)
+    @jokes[-1 + id]
```

Comment:
This could be more dynamic. Can you think of an enumerable that you can use to find a specific joke in the jokes array based on the id?